mqtt/mqtt-broker/src/session.rs

```diff
@@ -598,8 +632,7 @@ pub enum Session {
 }
 
 impl Session {
-    pub fn new_transient(auth_id: AuthId, connreq: ConnReq) -> Self {
-        let state = SessionState::new(connreq.client_id().clone());
+    pub fn new_transient(auth_id: AuthId, connreq: ConnReq, state: SessionState) -> Self {
```

Comment:
It looks weird that for the transient session you need to create a session state first. It seems like should be encapsulated in the `new_transient` method

Reply:
Or if you want to move it out and make it similar to other methods, you don't need to pass ConnReq inside, pass its parts instead

Reply:
I wanted to move it out since `SessionState` now requires config. Since `Broker` owns the config, I wanted to consolidate `SessionState` construction in the broker, instead of passing config into `new_transient`.